### PR TITLE
Add list of metrics

### PR DESCRIPTION
**Pull request by hugomendix**
Documenting all Mendix metrics explicitly.

### CODE REVIEW

File: content/en/docs/refguide/runtime/metrics.md [hunk 1]
```diff
@@ -382,6 +382,100 @@ Below are deprecated usages which will be removed in the future releases,
...
+Additionally, a suffix may be added for the unit of the metric (e.g. `_bytes`).
+{{% /alert %}}
+
+### 7.2 Additional Metrics
+
+#### 7.2.1 System and Process Metrics 
+
+| Name | Type | Description |
+| --- | --- | --- |
+| `system.cpu.count` | gauge | The number of processors available to the Java Virtual Machine. |
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/metrics.md:426]: Real processors or virtual processors?
Does it make a difference?

File: content/en/docs/refguide/runtime/metrics.md [hunk 2]
```diff
@@ -382,6 +382,100 @@ Below are deprecated usages which will be removed in the future releases,
...
+| `system.cpu.usage` | gauge | The recent CPU usage of the system the Runtime Server is running on (in the range [0..1]). |
+| `system.load.average.1m` | gauge | The average number of runnable (queued and running) threads on the available processors. |
+| `process.cpu.usage` | gauge | The recent CPU usage for the Java Virtual Machine process (in the range [0..1]). |
+
+#### 7.2.2 Java Virtual Machine Metrics 
+
+| Name | Type | Tags | Description |
+| --- | --- | --- | --- |
+| `jvm_info` | gauge | `vendor`, `runtime`, `version` | Details about the Java Virtual Machine (represented in the tags, the value is always 1). |
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/metrics.md:435]: The value of what is always `1`? Not sure what this means.

File: content/en/docs/refguide/runtime/metrics.md [hunk 3]
```diff
@@ -382,6 +382,100 @@ Below are deprecated usages which will be removed in the future releases,
...
+| `jvm.classes.loaded` | gauge | | The number of classes that are currently loaded in the Java Virtual Machine. |
+| `jvm.classes.unloaded` | counter | | The total number of classes unloaded since the Java Virtual Machine has started execution. |
+| `jvm.buffer.count` | gauge | `id` | An estimate of the number of buffers in a Java Virtual Machine memory pool (`id`). |
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/metrics.md:438]: memory pool or **buffer** pool?
I assume you mean that you can identify which buffer using `id`?

File: content/en/docs/refguide/runtime/metrics.md [hunk 4]
```diff
@@ -382,6 +382,100 @@ Below are deprecated usages which will be removed in the future releases,
...
+| `jvm.buffer.memory_used` | gauge | `id` | An estimate of the memory (in bytes) that the Java Virtual Machine is using for a buffer pool (`id`). |
+| `jvm.buffer.total_capacity` | gauge | `id` | An estimate of the total capacity (in bytes) of the buffers in a buffer pool (`id`). |
+| `jvm.memory.used` | gauge | `id`, `area` | The amount of memory (in bytes) used for a certain area (`area`): `heap` or `nonheap`. |
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/metrics.md:441]: What is `id` here?

File: content/en/docs/refguide/runtime/metrics.md [hunk 5]
```diff
@@ -382,6 +382,101 @@ Below are deprecated usages which will be removed in the future releases,
...
+| `mx.runtime.stats.connectionbus.deletes` | counter | `XASId` | The total number of `DELETE` statements that were executed on the database by a node (`XASId`) since it was started. |
+| `mx.odata.retrieve` | counter | `entity` | The total number of OData requests that were sent for a particular type of object (`entity`). |
+| `mx.odata.publish.objects` | counter | `entity` | The total number of OData objects that were served for a particular type of object (`entity`). |
+
+{{% alert color="info" %}}
+Note that the actual name may vary slightly depending on the back-end (for example, Prometheus replaces dots by underscores).
+Additionally, a suffix may be added for the unit of the metric (for example, `_bytes`).
+{{% /alert %}}
+
+### 7.2 Additional Metrics
```
Comment (MarkvanMents) [inline content/en/docs/refguide/runtime/metrics.md:416]: Are any of these especially useful to the customer, or do we just leave it to the customer to decide what they want to look at?

**Review by MarkvanMents — APPROVED** [review]
Hi Hugo (@hugomendix)
Looks like some great information here.
I have asked a couple of questions, but the PR is ready to go if you don't think anything needs changing.

Let me know if I can publish.

Thanks for the contribution.

Mark

File: content/en/docs/refguide/runtime/metrics.md (hunk 1 above)
Comment (hugomendix) [reply content/en/docs/refguide/runtime/metrics.md:426]: It doesn't make a difference, I think. It's the number of processors on the system that is hosting the JVM. If that's a virtual machine then it's virtual processors and if it's bare metal then it's real processor (cores).

File: content/en/docs/refguide/runtime/metrics.md (hunk 2 above)
Comment (hugomendix) [reply content/en/docs/refguide/runtime/metrics.md:435]: That this is a gauge that does not change. It's numeric value is irrelevant and always 1. The relevant information (which also doesn't change during execution) is in the tags of the gauge. In actual fact, these are just some string constants identifying the JVM.

File: content/en/docs/refguide/runtime/metrics.md (hunk 3 above)
Comment (hugomendix) [reply content/en/docs/refguide/runtime/metrics.md:438]: I copied this verbatim from the Micrometer documentation. The code effectively calls some Java MX bean: https://docs.oracle.com/en/java/javase/11/docs/api/java.management/java/lang/management/BufferPoolMXBean.html#getCount()

Unfortunately, the documentation for that isn't much clearer:
> Returns an estimate of the number of buffers in the pool.

Comment (hugomendix) [reply content/en/docs/refguide/runtime/metrics.md:438]: The `id` tag is the name of the buffer pool. From the same page:

> Returns the name representing this buffer pool.

File: content/en/docs/refguide/runtime/metrics.md (hunk 4 above)
Comment (hugomendix) [reply content/en/docs/refguide/runtime/metrics.md:441]: The name of the buffer pool, as used by the JVM.

File: content/en/docs/refguide/runtime/metrics.md (hunk 5 above)
Comment (hugomendix) [reply content/en/docs/refguide/runtime/metrics.md:416]: No, I don't think they're very useful to the customer. They're simply implementation details that leak out through the usage of the Micrometer library.

**Comment by hugomendix** [comment]
AFAIC you can publish this, unless you want to make some change in response to my asnwers.

File: content/en/docs/refguide/runtime/metrics.md (hunk 2 above)
Comment (MarkvanMents) [reply content/en/docs/refguide/runtime/metrics.md:435]: OK - makes sense.

**Comment by MarkvanMents** [comment]
> AFAIC you can publish this, unless you want to make some change in response to my asnwers.

Thanks for the answers. I don't think I need to change anything. If the customer is looking at this, we have to assume a reasonable amount of knowledge (i.e. more than I have). I'll publish this now.

**Pull request merged**